contracts/src/7683/T1ERC7683.sol

```diff
@@ -105,11 +123,46 @@ contract T1ERC7683 is BasicSwap7683, OwnableUpgradeable {
     /// @dev Decodes the message and processes settlement or refund operations accordingly
     /// @param _messageOrigin The domain from which the message originates
     /// @param _messageSender The address of the sender on the origin domain
+    /// @param _batchIndex The index of the batch where the PoF belongs to
+    /// @param _proofOfFillMerkleProof The proof of fill merkle root
+    /// @param _nonce The nonce of the message
     /// @param _message The encoded message received via t1
-    function _handle(uint32 _messageOrigin, bytes32 _messageSender, bytes calldata _message) internal {
+    function _handle(
+        uint32 _messageOrigin,
+        bytes32 _messageSender,
+        uint256 _batchIndex,
+        bytes calldata _proofOfFillMerkleProof,
+        uint256 _nonce,
+        bytes calldata _message
+    )
+        internal
+    {
+        IT1Chain t1Chain = IT1Chain(IL1T1Messenger(address(messenger)).rollup());
+
+        // Check if intent proof batch index is finalized
+        if (!t1Chain.isBatchFinalized(_batchIndex)) revert BatchNotFinalized();
+
+        bytes32 _proofOfFillRoot = t1Chain.proofOfFill7683Roots(_batchIndex);
+
+        // Check intent proof exists in our rollup
+        if (_proofOfFillRoot != _proofOfFillRoot) {
+            revert IntentProofNotFound(_batchIndex, _proofOfFillRoot);
```

Comment:
To me this error message does not accurately describe why this error would be thrown